eth/handler.go

```diff
@@ -191,12 +198,24 @@ func (pm *ProtocolManager) Stop() {
 	// take between 5 to 10 seconds and therefor feedback is required.
 	glog.V(logger.Info).Infoln("Stopping ethereum protocol handler...")
 
-	pm.quit = true
 	pm.txSub.Unsubscribe()         // quits txBroadcastLoop
 	pm.minedBlockSub.Unsubscribe() // quits blockBroadcastLoop
-	close(pm.quitSync)             // quits syncer, fetcher, txsyncLoop
 
-	// Wait for any process action
+	// Quit the sync loop.
+	// After this send has completed, no new peers will be accepted.
+	pm.noMorePeers <- struct{}{}
```

Comment:
The idea is to be somewhat certain that no peers will be registered after the send.
closing is asynchronous, the loop could still receive from newPeerCh after it.